### PR TITLE
fix: 🐛 add a mechanism with shared secret to bypass auth check

### PR DESCRIPTION
**Pull request by severo**
Option 1 in
https://github.com/huggingface/datasets-server/issues/877#issuecomment-1456474625.

### CODE REVIEW

File: services/api/src/api/authentication.py [hunk 1]
```diff
@@ -42,12 +45,20 @@ def auth_check(
...
+        request (Request | None): the request which optionally bears authentication headers: "cookie",
+          "authorization" or "X-Api-Key"
+        external_auth_bypass_key (str|None): if the request bears this secret key in the "X-Api-Key" header, the
+          external authentication service is bypassed.
 
     Returns:
         None: the dataset is authorized for the request
     """
+    if (
+        external_auth_bypass_key is not None
```
Comment (coyotte508) [inline services/api/src/api/authentication.py:57]: In the README it says this: 

> Defaults to `""`.

But here it checks against `None`

**Review by coyotte508 — APPROVED** [review]

File: services/api/src/api/authentication.py (hunk 1 above)
Comment (coyotte508) [reply services/api/src/api/authentication.py:57]:
```suggestion
        external_auth_bypass_key
```
```suggestion
        not not external_auth_bypass_key
```
I feel like either would be slightly better, to guard against the env being set to an empty string as well

Comment (severo) [reply services/api/src/api/authentication.py]: You're right, thanks!

**Comment by severo** [comment]
See https://github.com/huggingface/datasets-server/pull/898, as an alternative: instead of comparing the value to a shared secret, we check if we can decode the value as a JWT with the public key

**Comment by severo** [comment]
Superseded by #898 (we decided to go with the JWT on the moonlanding side)

**Pull request closed without merging**